### PR TITLE
server,engine-schema: fix listing service offering for vm scale

### LINKED ISSUE #7389: UI: scale VM not able to use same custom offering
##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.18.0.0
~~~

##### SUMMARY

You should be able to use the same compute offering for scaling a VM when the offering is a custom offering with variable cpu and memory values. Via api call scaleVirtualMachine it is possible to use the same serviceofferingid with other values via details parameter.

##### STEPS TO REPRODUCE

~~~
Create a custom compute offering. Create a new VM using this offering. Stop VM and click on the scale VM icon.
~~~

##### EXPECTED RESULTS

~~~
You should be able to edit cpu an memory of the VM by using the same custom offering
~~~

##### ACTUAL RESULTS

~~~
Custom offering is not visible in UI.
~~~

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Fixes #7389

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

Steps
- Create a custom compute offering with `disk_offering_strictness=true`
- Deploy a VM using it
- Stop the VM
- Try to scale VM using UI

Without fix - all other supported offerings are listed UI
With fix - only those supported offerings are listed in UI which use the same disk offering as the current one

### CODE REVIEW

**Comment by shwstppr** [comment]
@harikrishna-patnala  cc @sbrueseke this PR fixes the listing of the current custom offering but now it just lists offerings that use the same disk offering when disk_offering_strictness=true. Is this correct behaviour?

Screenshot from 2023-04-04 16-08-31

**Comment by harikrishna-patnala** [comment]
yes @shwstppr I think we are enforcing this service layer, it can be while listing the offering as well.
Since disk offering strictness is true then root disk offering cannot be changed, so makes sense

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan test

File: framework/quota/src/main/java/org/apache/cloudstack/quota/vo/ServiceOfferingVO.java [hunk 1]
```diff
@@ -149,6 +149,7 @@ public ServiceOfferingVO(ServiceOfferingVO offering) {
         vmType = offering.getSystemVmType();
         systemUse = offering.isSystemUse();
         dynamicScalingEnabled = offering.isDynamicScalingEnabled();
+        diskOfferingStrictness = offering.diskOfferingStrictness;
```
Comment (DaanHoogland) [inline framework/quota/src/main/java/org/apache/cloudstack/quota/vo/ServiceOfferingVO.java:152]: I think this file should be removed and not maintained, It pertains to the same table as https://github.com/apache/cloudstack/pull/7398/files#diff-bbbf939d866dbaec16b3c9ce76664ab203f7ba2a914e29772f795dd44a2cf993 and does not add value.

Comment (shwstppr) [reply framework/quota/src/main/java/org/apache/cloudstack/quota/vo/ServiceOfferingVO.java:152]: @DaanHoogland will create a cleanup PR for it

**Review by rohityadavcloud — APPROVED** [review]

**Review by weizhouapache — APPROVED** [review]
code lgtm

@shwstppr 
can you add the steps to reproduce the issue in the description ?

**Comment by shwstppr** [comment]
@weizhouapache done

File: framework/quota/src/main/java/org/apache/cloudstack/quota/vo/ServiceOfferingVO.java (hunk 1 above)
Comment (shwstppr) [reply framework/quota/src/main/java/org/apache/cloudstack/quota/vo/ServiceOfferingVO.java:152]: @DaanHoogland created a PR here, https://github.com/apache/cloudstack/pull/7439

**Review by DaanHoogland — APPROVED** [review]
looks good

**Comment by sbrueseke** [comment]
it looks like this fix was left out of 4.19.1.1?!?

**Pull request merged**